lib/nostr.js

```diff
@@ -17,155 +15,148 @@ export const DEFAULT_CROSSPOSTING_RELAYS = [
   'wss://nostr.mutinywallet.com/',
   'wss://relay.mutinywallet.com/'
 ]
-
-export class Relay {
-  constructor (relayUrl) {
-    const ws = new WebSocket(relayUrl)
-
-    ws.onmessage = (msg) => {
-      const [type, notice] = JSON.parse(msg.data)
-      if (type === 'NOTICE') {
-        console.log('relay notice:', notice)
-      }
-    }
-
-    ws.onerror = (err) => {
-      console.error('websocket error:', err.message)
-      this.error = err.message
-    }
-
-    this.ws = ws
-    this.url = relayUrl
-    this.error = null
-  }
-
-  static async connect (url, { timeout } = {}) {
-    const relay = new Relay(url)
-    await relay.waitUntilConnected({ timeout })
-    return relay
-  }
-
-  get connected () {
-    return this.ws.readyState === WebSocket.OPEN
+export const RELAYS_BLACKLIST = []
+
+/**
+ * @import {NDKSigner} from '@nostr-dev-kit/ndk'
+ * @import { NDK } from '@nostr-dev-kit/ndk'
+ * @import {NDKNwc} from '@nostr-dev-kit/ndk'
+ * @typedef {Object} Nostr
+ * @property {NDK} ndk
+ * @property {function(string, {logger: Object}): Promise<NDKNwc>} nwc
+ * @property {function(Object, {privKey: string, signer: NDKSigner}): Promise<NDKEvent>} sign
+ * @property {function(Object, {relays: Array<string>, privKey: string, signer: NDKSigner}): Promise<NDKEvent>} publish
+ */
+export class Nostr {
+  /**
+   * @type {NDK}
+   */
+  _ndk = null
+
+  constructor ({ privKey, defaultSigner, relays, supportNip07 = false, ...ndkOptions } = {}) {
+    this._ndk = new NDK({
+      explicitRelayUrls: relays,
+      blacklistRelayUrls: RELAYS_BLACKLIST,
+      autoConnectUserRelays: false,
+      autoFetchUserMutelist: false,
+      clientName: 'stacker.news',
+      signer: defaultSigner ?? this.getSigner({ privKey, supportNip07 }),
+      ...ndkOptions
+    })
   }
 
-  get closed () {
-    return this.ws.readyState === WebSocket.CLOSING || this.ws.readyState === WebSocket.CLOSED
+  /**
+   * @type {NDK}
+   */
+  get ndk () {
+    return this._ndk
   }
 
-  async waitUntilConnected ({ timeout } = {}) {
-    let interval
-
-    const checkPromise = new Promise((resolve, reject) => {
-      interval = setInterval(() => {
-        if (this.connected) {
-          resolve()
-        }
-        if (this.closed) {
-          reject(new Error(`failed to connect to ${this.url}: ` + this.error))
-        }
-      }, 100)
-    })
-
-    try {
-      return await withTimeout(checkPromise, timeout)
-    } catch (err) {
-      this.close()
-      throw err
-    } finally {
-      clearInterval(interval)
-    }
+  /**
+   *
+   * @param {args} param0
+   * @param {string} [args.privKey] - private key to use for signing
+   * @param {boolean} [args.supportNip07] - whether to use NIP-07 signer if available
+   * @returns {NDKPrivateKeySigner | NDKNip46Signer | NDKNip07Signer | null} - a signer instance
```

Comment:
nitpick: `| NDKNip46Signer` should have been part of #1636 and not included here already

but it's okay now